### DISCUSSION TITLE
Function Times Out

### DISCUSSION THREAD

**Discussion by ShowMeTheStack** [appwrite/appwrite#3917] [Q&A] [answered]
We created a Dart 2.17 runtime based cloud function. The code for the function is as follows:

```
import 'package:dart_appwrite/dart_appwrite.dart';
import 'package:dart_appwrite/models.dart';

Future<void> start(final req, final res) async {
  Client client = Client();
  Users users = Users(client);

  if (req.variables['APPWRITE_FUNCTION_ENDPOINT'] == null ||
      req.variables['APPWRITE_FUNCTION_API_KEY'] == null) {
    print(
        "Environment variables are not set. Function cannot use Appwrite SDK.");
  } else {
    client
        .setEndpoint(req.variables['APPWRITE_FUNCTION_ENDPOINT'])
        .setProject(req.variables['APPWRITE_FUNCTION_PROJECT_ID'])
        .setKey(req.variables['APPWRITE_FUNCTION_API_KEY'])
        .setSelfSigned(status: true);
  }
  res.json({
    'areDevelopersAwesome': true,
  });
  
  User user = await users.get(userId: "632c1697c94ea0fbc5dc");
  print(user.email);
}

```

Without the await for the future returned by `users.get`, the code terminates in about 15 seconds (which is another issue that I'll write, the future in my server code that invokes the cloud function using `functions.createExecution` takes 15 seconds to complete regardless of the code in the cloud function).

With the await, the code times out with an empty `res`.

**Comment by damalo93** [comment]
I have the same problem

**Comment by stnguyen90** [comment, marked as answer]
@ShowMeTheStack, what did you set for your `APPWRITE_FUNCTION_ENDPOINT` variable?

**Reply by stnguyen90** [reply]
> That environment variable is set to http://localhost/v1

@ShowMeTheStack, for Appwrite Functions, you can't use localhost for the endpoint because the runtime container will try to connect to itself rather than Appwrite. Please try using the LAN IP of your machine instead.

**Comment by ShowMeTheStack** [comment]
That environment variable is set to http://localhost/v1
